content/post/blackboard_submission_service.md

```diff
@@ -0,0 +1,105 @@
++++
+date = "2015-01-19T19:50:00+10:00"
+draft = true
+title = "Blackboard Extensions and the Submission Service"
+description = "Stepping through the process for adding the required fields to your bb-manifest file."
+keywords = ["blackboard"]
++++
+
+# Blackboard Extensions and the Submission Service
+
+Extensions are one of many Blackboard APIs, and like the rest of them, are under-documented or have very limited implementation details. This post will walk through the process for implementing one such extension provided by Blackboard, namely the submission service. The submission service provides event handlers for submittable items (assignments, tests, et.al). For example when a student submits an assignment, one of the handlers in this extension fires. So let's jump into the code!
+
+## Quick Setup
+
+Start by grabbing the jar file, it can be obtained from the Blackboard development vm. For the submission service, the jar file is located at `/usr/local/blackboard/content/vi/BBLEARN/plugins/bb-submission-services/webapp/WEB-INF/libext/submission-services-api-1.1-SNAPSHOT.jar`. Move this file onto the class path for your building block.
+
+## bb-manifest
+
+Implementing Blackboard extensions also requires some modifications to the `bb-manifest.xml` file. This building block requires extension points, so start by adding `<webapp-type value="javaext" />` as explained [in these docs](https://docs.alltheducks.com/blackboard/bb-manifest-ref.html#toc_18).
+
+For our service to be registered by blackboard, we must add a requirement for it:
```

Comment:
"To have the Submission Service building block's libraries added to you building block's classpath at run time, you need to specify it as a dependency."